contracts/internal/EverlongPositions.sol

```diff
@@ -0,0 +1,295 @@
+        // TODO: Enable closing of mature positions incrementally to avoid
+        //       the case where the # of mature positions exceeds the max
+        //       gas per block.
+        while (hasMaturedPositions()) {
+            // Retrieve the oldest matured position and close it.
+            _position = getPosition(0);
+            IHyperdrive(_hyperdrive).closeLong(
+                _position.maturityTime,
+                _position.bondAmount,
+                _minCloseLongOutput(
+                    _position.maturityTime,
+                    _position.bondAmount
+                ),
+                IHyperdrive.Options(address(this), _asBase, "")
+            );
+
+            // Update positions to reflect the newly closed long.
+            _handleCloseLong(uint128(_position.bondAmount));
+        }
+    }
+
+    /// @dev Account for closed bonds at the oldest `maturityTime`
+    ///      within the `PositionManager`.
+    /// @param _bondAmountClosed Amount of bonds closed.
+    function _handleCloseLong(uint128 _bondAmountClosed) internal {
+        // Remove the oldest position from the front queue.
+        Position memory _position = _decodePosition(_positions.popFront());
+
+        // Compare the input bond amount
+        // to the most mature position's `bondAmount`.
+        if (_bondAmountClosed > _position.bondAmount) {
+            revert InconsistentPositionBondAmount();
+        } else if (_bondAmountClosed == _position.bondAmount) {
+            // The amount to close equals the position size.
+            // Nothing further needs to be done.
+            emit PositionClosed(_position.maturityTime);
+        } else {
+            // The amount to close is not equal to the position size.
+            // Push the position less the amount of longs closed to the front.
+            _positions.pushFront(
+                _encodePosition(
+                    _position.maturityTime,
+                    _position.bondAmount - _bondAmountClosed
+                )
+            );
+            emit PositionUpdated(
+                _position.maturityTime,
+                _position.bondAmount - _bondAmountClosed,
+                0
+            );
+        }
+    }
+
+    // ╭─────────────────────────────────────────────────────────╮
+    // │ Position Encoding/Decoding                              │
+    // ╰─────────────────────────────────────────────────────────╯
+
+    /// @dev Encodes the `_maturityTime` and `_bondAmount` into bytes32
+    ///      to store in the queue.
+    /// @param _maturityTime Timestamp the position matures.
+    /// @param _bondAmount Amount of bonds in the position.
+    /// @return The bytes32 encoded `Position`.
+    function _encodePosition(
+        uint128 _maturityTime,
+        uint128 _bondAmount
+    ) public pure returns (bytes32) {
+        return
+            (bytes32(bytes16(_maturityTime)) >> 128) |
+            bytes32(bytes16(_bondAmount));
```

Comment:
Good catch! For some reason I thought `uint128` needed to be converted to `bytes16` before `bytes32` but totally isn't the case

Reply:
Sounds good, and totally makes sense. In case it's helpful, my normal route to get from sized `uint` types to get to `bytes32` (or other bytes types) is to widen them until casting from `uint256` to `bytes32` is as trivial as telling Solidity what type it is (and nothing significant happens under the hood). It's exactly the opposite when I get to bytes, where I like to go from `bytes32` to narrow to whatever type I want to get to. This will leave the bytes in place and will just take the slice that I want.